### ISSUE TITLE
Reboot after watchdogctl disable

### ISSUE THREAD

**Issue by pietercardoen** [troglobit/watchdogd#16]
Test sequence:
* watchdogd --loglevel=info -n  => keep the log of this.
* watchdogctl disable
* Start application which fails to meet the kick deadline
* After 30 seconds the system reboots while the watchdog should be disabled.

Logging of watchdogd:

watchdogd[992]: Starting process supervisor, waiting for client subscribe ...
watchdogd[992]: watchdogd v3.1 starting ...
watchdogd[992]: /dev/watchdog: imx2+ watchdog, capabilities 0x8390
watchdogd[992]: Reset cause: WATCHDOG
watchdogd[992]: Kicking WDT.
watchdogd[992]: Kicking WDT.
watchdogd[992]: Attempting to disable HW watchdog timer.
watchdogd[992]: Process ListenerUDP::x1:239.23.255.5:50501[1020] failed to meet its deadline, rebooting ...
watchdogd[992]: Waiting for HW WDT reboot ...

watchdogd[992]: HW WDT did not reboot, forcing reboot now ...

**Comment by troglobit** [comment]
watchdogd can only do half of the job here.  It writes 'V' to the driver, but depending on the kernel CONFIG_WATCHDOG_NOWAYOUT and the driver parameter nowayout the driver either honors the disable command, or not.

**Comment by pietercardoen** [comment]
We have indeed configured CONFIG_WATCHDOG_NOWAYOUT.

However the behaviour is different compared to a regular watchdog reset. It behaves now as a SW reset (like reboot). The reset cause is different.

**Comment by troglobit** [comment]
Now that might be a bug in watchdogd, but I'm pretty sure the reboot is caused by the WDT driver.

**Comment by troglobit** [comment]
The more I think of it, this case may be undefined. What requirement should we set for this case, what reset cause should be saved (prepared) when the system is restarted and watchdogd is disabled?

**Comment by pietercardoen** [comment]
On monday, I will continue investigating the behaviour as I now have no hardware available.

On your question: I prefer that the system doesn't restart when watchdogd is disabled. We have currently a malfunctioning application which  doesn't unsubscribe. We were testing with watchdogd disabled and now we had no reset but a SW reset. It would be good to just do nothing when the watchdogd is disabled.

**Comment by troglobit** [comment]
How did your investigation go, should we close this issue?

**Comment by pietercardoen** [comment]
I have got time to have a look at this issue. In my opinion, this is a bug.

In normal operation, the watchdog should reset the device and when the watchdog doesn't work for some reason, a reboot is applied. This is a good practice to have a fallback.

However if the watchdog is disabled, it still applies the reboot. I changed the behaviour to exit the watchdog daemon if it's disabled without rebooting.
